artemis/modules/subdomain_enumeration.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/env python3
+import time
+from typing import Optional, Set, Callable
+
+from karton.core import Task
+
+from artemis.binds import TaskStatus, TaskType
+from artemis.config import Config
+from artemis.domains import is_subdomain
+from artemis.module_base import ArtemisBase
+from artemis.utils import check_output_log_on_error
+
+DOMAIN_REGEX = r"([a-z0-9\-]+\.)+[a-z0-9\-]+"
+
+# NOTE: The rappidns, jldc, gau and crtsh modules were removed from this class 
+# as their functionality is already implemented internally by the subfinder and amass utilities.
+class UnableToObtainSubdomainsException(Exception):
+    pass
+
+class SubdomainEnumeration(ArtemisBase):
+    """
+    Consumes `type: domain` to gather subdomains and produces `type: domain`.
+    """
+
+    identity = "SubdomainEnumeration"
+    filters = [
+        {"type": TaskType.DOMAIN.value},
+    ]
+    lock_target = False
+
+    def get_subdomains_with_retry(self, func: Callable[[str], Optional[Set[str]]], domain: str, retries: int = 20, sleep_time: int = 300) -> Set[str]:
+        for retry_id in range(retries):
+            subdomains = func(domain)
+            if subdomains is not None:
+                return subdomains
+
+            self.log.info("Retrying for domain %s", domain)
+            if retry_id < retries - 1:
+                time.sleep(sleep_time)
+            else:
+                raise UnableToObtainSubdomainsException(f"Unable to obtain subdomains for {domain} after {retries} retries")
+        
+        return set()
+
+    def get_subdomains_from_subfinder(self, domain: str) -> Optional[Set[str]]:
```

Comment:
I insist on adding gau, as the list you provided from cert.pl skipped some domains that gau gives us (e.g. nomoreransom.cert.pl)